source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -784,17 +808,48 @@ inline void ComponentInterface<NodeT>::set_predicate(
 }
 
 template<class NodeT>
-inline std::string ComponentInterface<NodeT>::validate_input_signal_name(const std::string& signal_name) {
+inline void ComponentInterface<NodeT>::declare_signal(
+    const std::string& signal_name, const std::string& type, const std::string& default_topic, bool fixed_topic
+) {
   std::string parsed_signal_name = utilities::parse_topic_name(signal_name);
   if (parsed_signal_name.empty()) {
     throw exceptions::AddSignalException(
-        "The parsed signal name for input '" + signal_name
+        "The parsed signal name for " + type + " '" + signal_name
             + "' is empty. Provide a string with valid characters for the signal name ([a-zA-Z0-9_]).");
   }
   if (this->inputs_.find(parsed_signal_name) != this->inputs_.cend()) {
-    throw exceptions::AddSignalException("Input with name '" + parsed_signal_name + "' already exists.");
+    throw exceptions::AddSignalException("Signal with name '" + parsed_signal_name + "' already exists as input.");
+  }
+  if (this->outputs_.find(parsed_signal_name) != this->outputs_.cend()) {
+    throw exceptions::AddSignalException("Signal with name '" + parsed_signal_name + "' already exists as output.");
+  }
+  std::string topic_name = default_topic.empty() ? "~/" + parsed_signal_name : default_topic;
+  auto parameter_name = parsed_signal_name + "_topic";
+  // TODO is the is_empty condition still valid?
```

Comment:
I'm not sure what to do here, mostly because I can't remember why we put the is_empty condition in the first place. But I think it should be removed, I dont see a reason why we should only set the parameter value if its empty...